packages/grid_client/tests/modules/workload.test.ts

```diff
@@ -0,0 +1,194 @@
+import { plainToClass } from "class-transformer";
+
+import { DeploymentResult, ResultStates, Workload, WorkloadTypes } from "../../src";
+import {
+  ComputeCapacity,
+  GatewayFQDNProxy,
+  GatewayNameProxy,
+  Mount,
+  PublicIP,
+  QuantumSafeFS,
+  Volume,
+  Zdb,
+  Zlogs,
+  Zmachine,
+  ZmachineNetwork,
+  Zmount,
+  Znet,
+} from "../../src";
+
+let workload: Workload;
+
+// Creates instances of workload data types and set their data
+const createDataInstance = (type: WorkloadTypes) => {
+  let instance;
+  const network = new ZmachineNetwork();
+  const computeCapacity = new ComputeCapacity();
+  const disks = new Mount();
+
+  const rootfs_size = 2;
+
+  switch (type) {
+    case WorkloadTypes.zmachine:
+      instance = new Zmachine();
+      computeCapacity.cpu = 1;
+      computeCapacity.memory = 256 * 1024 ** 2;
```

Comment:
might be a good idea to declare 1024 as a constant as it is used more than once

Reply:
> might be a good idea to declare 1024 as a constant as it is used more than once

I think this's not needed as I only used twice and the 1024 indicates the number of bytes